web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx

```diff
@@ -138,9 +142,15 @@ export function EnrollEksCluster(props: AgentStepProps) {
   // join token will be set only if user opens ManualHelmDialog,
   // we delay it to avoid premature admin action MFA confirmation request.
   const [joinToken, setJoinToken] = useState<JoinToken>(null);
+  const [customLabels, setCustomLabels] = useState<ResourceLabel[]>([]);
 
   const ctx = useTeleport();
 
+  useEffect(() => {
+    // when changing selected cluster, clear defined labels
+    setCustomLabels([]);
+  }, [selectedCluster]);
```

Comment:
`selectedCluster` and `setSelectedCluster` belong to this component. Instead of using an effect to do this, it's better to create a function that's going to both set selected cluster _and_ clear defined labels and then use it instead of `setSelectedCluster`.

https://react.dev/learn/you-might-not-need-an-effect#sharing-logic-between-event-handlers
https://react.dev/learn/you-might-not-need-an-effect#chains-of-computations